packages/fxa-event-broker/lib/serviceNotifications.ts

```diff
@@ -0,0 +1,126 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { Logger } from 'mozlog';
+import * as joi from 'typesafe-joi';
+
+// Event strings
+export const LOGIN_EVENT = 'login';
+export const SUBSCRIPTION_UPDATE_EVENT = 'subscription:update';
+export const DELETE_EVENT = 'delete';
+
+// Message schemas
+const BASE_MESSAGE_SCHEMA = joi
+  .object()
+  .keys({
+    event: joi.string().required()
+  })
+  .unknown(true)
+  .required();
+
+const LOGIN_SCHEMA = joi
+  .object()
+  .keys({
+    clientId: joi.string().optional(),
+    deviceCount: joi
+      .number()
+      .integer()
+      .required(),
+    email: joi.string().required(),
+    event: joi
+      .string()
+      .valid(LOGIN_EVENT)
+      .required(),
+    service: joi.string().required(),
+    uid: joi.string().required(),
+    userAgent: joi.string().required()
+  })
+  .unknown(true)
+  .required();
+
+const SUBSCRIPTION_UPDATE_SCHEMA = joi
+  .object()
+  .keys({
+    event: joi
+      .string()
+      .valid(SUBSCRIPTION_UPDATE_EVENT)
+      .required(),
+    eventCreatedAt: joi
+      .number()
+      .integer()
+      .required(),
```

Comment:
How about using [joi's timestamp validator](https://github.com/hapijs/joi/blob/v9.0.0-3/API.md#datetimestamptype) here, to ensure we only accept millisecond-based timestamps? `Joi.date().timestamp('javascript')`

Reply:
The number here is actually in seconds (its Stripe's event timestamp). I'm actually hoping to get rid of all these joi schemas we have floating about and use JSON-Schema, but that's coming up. 😄